src/v/cluster/cluster_recovery_state.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "cluster/cloud_metadata/cluster_manifest.h"
+#include "cluster/types.h"
+#include "model/fundamental.h"
+#include "serde/envelope.h"
+#include "serde/serde.h"
+
+namespace cluster {
+
+// Tracks the state of an on-going cluster recovery. This only exposes an
+// interface to expose and modify the current status of recovery; an external
+// caller should use these to actually drive recovery.
+//
+// This only tracks the parts of recovery that are not already tracked by
+// another subsystem of the controller. I.e. topics, configs, etc. are tracked
+// in other controller tables; but consumer offsets pending recovery are
+// tracked here.
+struct cluster_recovery_state
+  : public serde::envelope<
+      cluster_recovery_state,
+      serde::version<0>,
+      serde::compat_version<0>> {
+public:
+    cluster_recovery_state() = default;
+    explicit cluster_recovery_state(
+      cloud_metadata::cluster_metadata_manifest manifest,
+      cloud_storage_clients::bucket_name bucket)
+      : manifest(std::move(manifest))
+      , bucket(std::move(bucket)) {}
+    auto serde_fields() { return std::tie(manifest, bucket, error_msg); }
```

Comment:
it's helpful to mark the fields below which are explicitly not serialized.

Reply:
Nice catch! This wasn't intentional. Will add a test in a subsequent PR that would catch this (restarting post recovery + snapshot).